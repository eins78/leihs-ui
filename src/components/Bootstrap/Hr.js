import React from 'react'
import cx from 'classnames'

const Hr = ({ children, ...p }) =>
  !children ? (
    <hr {...p} />
  ) : (
    <div {...p} className={cx(p.className, 'hr-text')}>
      <hr />
      {children}
    </div>
  )

export default Hr
