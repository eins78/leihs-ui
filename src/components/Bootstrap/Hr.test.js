import React from 'react'
import renderer from 'react-test-renderer'

import Hr from './Hr'

const examples = [
  {
    title: 'without inner text',
    content: (
      <div>
        <Hr />
      </div>
    )
  },
  {
    title: 'with inner text',
    content: (
      <div>
        <Hr>Inner Text</Hr>
      </div>
    )
  }
]

examples.forEach(({ name, content }) => {
  it(`${name}: renders correctly`, () => {
    expect(renderer.create(content).toJSON()).toMatchSnapshot()
  })
})
