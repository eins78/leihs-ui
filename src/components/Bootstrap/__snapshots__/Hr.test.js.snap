// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`undefined: renders correctly 1`] = `
<div>
  <hr />
</div>
`;

exports[`undefined: renders correctly 2`] = `
<div>
  <div
    className="hr-text"
  >
    <hr />
    Inner Text
  </div>
</div>
`;
